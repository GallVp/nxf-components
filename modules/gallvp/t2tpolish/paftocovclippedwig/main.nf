process T2TPOLISH_PAFTOCOVCLIPPEDWIG {
    tag "${meta.id}"
    label 'process_single'

    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    container "ghcr.io/gallvp/t2t-polish:sha-f780afa"

    input:
    tuple val(meta), path(paf)
    val name
    val span

    output:
    tuple val(meta), path("*.clip_abs.wig"), emit: clip_abs
    tuple val(meta), path("*.clip_norm.wig"), emit: clip_norm
    tuple val(meta), path("*.cov.wig"), emit: cov
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def avail_mem = (task.memory.giga * 0.8).intValue()
    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def VERSION = 'f780afa'
    """
    java \\
        -Xmx${avail_mem}g \\
        -jar \\
        \$T2T_POLISH/paf_util/pafToCovClippedWig.jar \\
        ${paf} \\
        '${name}' \\
        ${span} \\
        ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        t2tpolish: ${VERSION}
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def avail_mem = (task.memory.giga * 0.8).intValue()
    // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    def VERSION = 'f780afa'
    """
    echo ${args}
    echo '-Xmx${avail_mem}g'

    touch ${prefix}.clip_abs.wig
    touch ${prefix}.clip_norm.wig
    touch ${prefix}.cov.wig

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        t2tpolish: ${VERSION}
    END_VERSIONS
    """
}
