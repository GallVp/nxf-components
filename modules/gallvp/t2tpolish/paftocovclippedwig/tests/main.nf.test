
nextflow_process {

    name "Test Process T2TPOLISH_PAFTOCOVCLIPPEDWIG"
    script "../main.nf"
    process "T2TPOLISH_PAFTOCOVCLIPPEDWIG"

    tag "modules"
    tag "modules_gallvp"
    tag "t2tpolish"
    tag "t2tpolish/paftocovclippedwig"

    test("sarscov2 - genome - paf") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.paf', checkIfExists: true)
                ]
                input[1] = 'Test track'
                input[2] = '128'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - genome - paf - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.paf', checkIfExists: true)
                ]
                input[1] = 'Test track'
                input[2] = '128'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}