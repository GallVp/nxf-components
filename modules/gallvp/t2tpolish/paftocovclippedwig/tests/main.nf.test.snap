{
    "sarscov2 - genome - paf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_abs.wig:md5,06c0564b7e570c76598fdee2daa44352"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_norm.wig:md5,a8030168e52d108d3e7d25fd796e65e7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cov.wig:md5,dd02a0f4e4fe110158367a7c7d0f665c"
                    ]
                ],
                "3": [
                    "versions.yml:md5,d5c9ffc91212d148d659ef46f9310241"
                ],
                "clip_abs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_abs.wig:md5,06c0564b7e570c76598fdee2daa44352"
                    ]
                ],
                "clip_norm": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_norm.wig:md5,a8030168e52d108d3e7d25fd796e65e7"
                    ]
                ],
                "cov": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cov.wig:md5,dd02a0f4e4fe110158367a7c7d0f665c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d5c9ffc91212d148d659ef46f9310241"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T13:56:33.494577"
    },
    "sarscov2 - genome - paf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_abs.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_norm.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cov.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,d5c9ffc91212d148d659ef46f9310241"
                ],
                "clip_abs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_abs.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "clip_norm": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip_norm.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cov": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cov.wig:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d5c9ffc91212d148d659ef46f9310241"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T13:56:37.274895"
    }
}